src/roles/_shaman_helper.py

```diff
@@ -0,0 +1,423 @@
+#
+# To add new shaman roles in your custom roles/whatever.py file:
+# 1. Expand var.TOTEM_ORDER and upate var.TOTEM_CHANCES to account for the new width
+# 2. Add the role to var.ROLE_GUIDE
+# 3. Add the role to whatever other holding vars are necessary based on what it does
+# 4. Implement custom events if the role does anything else beyond giving totems.
+#
+# Modifying this file to add new totems or new shaman roles is generally never required
+
+def _get_target(var, wrapper, message, lastgiven):
+    target = get_target(var, wrapper, re.split(" +", message)[0], allow_self=True)
+    if not target:
+        return
+
+    if lastgiven.get(wrapper.source) is target:
+        wrapper.send(messages["shaman_no_target_twice"].format(target))
+        return
+
+    return target
+
+def _give_totem(var, wrapper, target, prefix, tags, role, msg, shamans):
+
+    orig_target = target
+    orig_role = get_main_role(orig_target)
+
+    evt = Event("targeted_command", {"target": target, "misdirection": True, "exchange": True},
+                action = "give a totem{0} to".format(msg))
+
+    if not evt.dispatch(var, "totem", wrapper.source, target, frozenset(tags)):
+        return
+
+    target = evt.data["target"]
+    targrole = get_main_role(target)
+
+    wrapper.send(messages["shaman_success"].format(prefix, msg, orig_target))
+    shamans[wrapper.source] = UserList((target, orig_target))
+
+    debuglog("{0} ({1}) TOTEM: {2} ({3}) as {4} ({5})".format(wrapper.source, role, target, targrole, orig_target, orig_role))
+
+def _apply_totem(totem, shaman, victim):
+    if totem == "death": # this totem stacks
+        DEATH[shaman] = victim
+    elif totem == "protection": # this totem stacks
+        PROTECTION.append(victim)
+    elif totem == "revealing":
+        REVEALING.add(victim)
+    elif totem == "narcolepsy":
+        NARCOLEPSY.add(victim)
+    elif totem == "silence":
+        SILENCE.add(victim)
+    elif totem == "desperation":
+        DESPERATION.add(victim)
+    elif totem == "impatience": # this totem stacks
+        IMPATIENCE.append(victim)
+    elif totem == "pacifism": # this totem stacks
+        PACIFISM.append(victim)
+    elif totem == "influence":
+        INFLUENCE.add(victim)
+    elif totem == "exchange":
+        EXCHANGE.add(victim)
+    elif totem == "lycanthropy":
+        LYCANTHROPY.add(victim)
+    elif totem == "luck":
+        LUCK.add(victim)
+    elif totem == "pestilence":
+        PESTILENCE.add(victim)
+    elif totem == "retribution":
+        RETRIBUTION.add(victim)
+    elif totem == "misdirection":
+        MISDIRECTION.add(victim)
+    elif totem == "deceit":
+        DECEIT.add(victim)
+    # other totem types possibly handled in an earlier event,
+    # as such there is no else: clause here
+
+@event_listener("see", priority=10)
+def on_see(evt, var, nick, victim):
+    if (users._get(victim) in DECEIT) ^ (users._get(nick) in DECEIT): # FIXME
```

Comment:
You updated the see event to pass users... did you not actually test all of the totems?

Reply:
Right, I misread a thing. This is fine and doesn't need to be changed.